payments-core/src/main/java/com/stripe/android/payments/core/authentication/DefaultIntentAuthenticatorRegistry.kt

```diff
@@ -0,0 +1,147 @@
+import com.stripe.android.PaymentBrowserAuthStarter
+import com.stripe.android.PaymentRelayStarter
+import com.stripe.android.StripePaymentController
+import com.stripe.android.model.StripeIntent
+import com.stripe.android.networking.AnalyticsRequestExecutor
+import com.stripe.android.networking.AnalyticsRequestFactory
+import com.stripe.android.networking.StripeRepository
+import com.stripe.android.payments.PaymentFlowResult
+import com.stripe.android.payments.core.injection.AuthenticationModule
+import com.stripe.android.payments.core.injection.DaggerAuthenticationComponent
+import com.stripe.android.stripe3ds2.service.StripeThreeDs2Service
+import com.stripe.android.stripe3ds2.transaction.MessageVersionRegistry
+import com.stripe.android.view.AuthActivityStarter
+import javax.inject.Inject
+import kotlin.coroutines.CoroutineContext
+
+/**
+ * Default registry to provide look ups for [IntentAuthenticator].
+ * Should be only accessed through [DefaultIntentAuthenticatorRegistry.getInstance].
+ */
+internal class DefaultIntentAuthenticatorRegistry @Inject internal constructor() :
+    IntentAuthenticatorRegistry {
+
+    // default authenticators always packaged with the SDK
+    // TODO(ccen) register a singleton instance of each authenticator with registration API
+    @Inject
+    lateinit var webIntentAuthenticator: WebIntentAuthenticator
+
+    @Inject
+    lateinit var noOpIntentAuthenticator: NoOpIntentAuthenticator
+
+    // authenticators requiring a 3p SDK
+    // TODO(ccen) move them to a dedicated module
+    @Inject
+    lateinit var stripe3DS2Authenticator: Stripe3DS2Authenticator
+
+    override fun lookUp(stripeIntent: StripeIntent): IntentAuthenticator {
+        if (!stripeIntent.requiresAction()) {
+            return noOpIntentAuthenticator
+        }
+
+        when (val nextActionData = stripeIntent.nextActionData) {
+            is StripeIntent.NextActionData.SdkData.Use3DS2 -> {
+                return stripe3DS2Authenticator
+            }
+            is StripeIntent.NextActionData.SdkData.Use3DS1 -> {
+                // can only triggered when `use_stripe_sdk=true`
+                return webIntentAuthenticator
+            }
+            is StripeIntent.NextActionData.RedirectToUrl -> {
+                // can only triggered when `use_stripe_sdk=false`
+                return webIntentAuthenticator
+            }
+            is StripeIntent.NextActionData.AlipayRedirect -> {
+                return webIntentAuthenticator
+            }
+            is StripeIntent.NextActionData.DisplayOxxoDetails -> {
+                return webIntentAuthenticator.takeIf { nextActionData.hostedVoucherUrl != null }
+                    ?: noOpIntentAuthenticator
+            }
+            else -> return noOpIntentAuthenticator
+        }
+    }
+
+    companion object {
+        // Holding this single instance would in turn holds DaggerAuthenticationComponent instance,
+        // which keeps dagger injection graph live.
```

Comment:
tiny: "would in turn hold" (no s at end of holds)

Reply:
removed this comment as new we're creating a new instance each time